### PR TITLE
enable archive flag

### PR DESCRIPTION
**Pull request by XavierCheng1996**
#241
enable the `--archive` flag in `Add/Copy` command. So in `Add/Copy --archive src dst` command, we will maintain the ownership of the `src` file rather than change the ownership to `root`.

### CODE REVIEW

File: testdata/build-context/copy-archive/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM alpine as  phaseA
```
Comment (yiranwang52) [inline testdata/build-context/copy-archive/Dockerfile]: extra space

File: testdata/build-context/copy-archive/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+RUN  mkdir /hello
```
Comment (yiranwang52) [inline testdata/build-context/copy-archive/Dockerfile]: extra space

File: testdata/build-context/copy-archive/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+RUN mkdir /hello/world
+RUN  touch /hello/world/a.txt
```
Comment (yiranwang52) [inline testdata/build-context/copy-archive/Dockerfile]: extra space

File: testdata/build-context/copy-archive/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+RUN chown daemon:daemon /hello/world
+
+FROM  alpine
```
Comment (yiranwang52) [inline testdata/build-context/copy-archive/Dockerfile]: extra space

File: lib/parser/dockerfile/add_copy.go [hunk 5]
```diff
@@ -14,30 +14,59 @@
...
+				archiveCount++
+				preserveOwner = true
+			} else {
+				return nil, fmt.Errorf("archive flag format is wrong")
+			}
+		}
+	}
+
+	if archiveCount+chownCount >= 2 {
+		return nil, base.err(fmt.Errorf("arguments shouldn't contain more than one flag [--chown or --archive]"))
```
Comment (yiranwang52) [inline lib/parser/dockerfile/add_copy.go]: uppercase `argument`?

File: lib/builder/step/copy_step_test.go [hunk 6]
```diff
@@ -44,7 +44,7 @@ func abs(path string) string {
 func TestNewCopyStep(t *testing.T) {
```
Comment (yiranwang52) [inline lib/builder/step/copy_step_test.go]: I think there need to be a new unit test here.

Comment (XavierCheng1996) [reply lib/builder/step/copy_step_test.go]: don't think so... I think they are closely coupled just like `chown`

File: lib/parser/dockerfile/add_copy.go (hunk 5 above)
Comment (XavierCheng1996) [reply lib/parser/dockerfile/add_copy.go]: it should be `argument` here and I think we should keep it lowercase.

1.   keep consistent with `fmt.Errof` elsewhere.
2.   isn't it golang style to keep it lowercase?

**Pull request merged**